internal/promote/pull_requests.go

```diff
@@ -28,9 +30,12 @@ func OpenPullRequestWithRemovedPackages(client *github.Client, username, head, b
 
 // OpenPullRequestWithPromotedPackages method opens a PR against "base" branch with promoted packages.
 // Head is the branch containing the changes that will be added to the base branch.
-func OpenPullRequestWithPromotedPackages(client *github.Client, username, head, base, sourceStage, destinationStage string, promotedPackages PackageVersions) (string, error) {
+func OpenPullRequestWithPromotedPackages(client *github.Client, username, head, base, sourceStage, destinationStage string, promotedPackages PackageVersions, r *git.Repository) (string, error) {
```

Comment:
This is looking better!

Just one last thought. I think it's possible to leave `OpenPullRequestWithPromotedPackages` as is and let it be responsible only for building PR content (concatenating strings) and calling Github API. What do you think about moving `calculatePackageSignature` to the `signature.go` and creating a new struct, e.g. `SignedPackageVersions`, that later on can be passed here:

`func OpenPullRequestWithPromotedPackages(client *github.Client, username, head, base, sourceStage, destinationStage string, signedPromotedPackages SignedPackageVersions) (string, error) {`

or 

`func OpenPullRequestWithPromotedPackages(client *github.Client, username, head, base, sourceStage, destinationStage string, promotedPackages PackageVersions,  signatures PackageSignatures) (string, error) {`

Reply:
Yeah that makes sense. I'm going to go with
```
func OpenPullRequestWithPromotedPackages(client *github.Client, username, head, base, sourceStage, destinationStage string, signedPromotedPackages SignedPackageVersions) (string, error) {
```

and have the `SignedPackageVersions` struct be composed of a `PackageVersion`